### ISSUE TITLE
Link verification fails on cohost.org profile URLs

### ISSUE THREAD

**Issue by eloquence** [mastodon/mastodon#27597]
### Steps to reproduce the problem

1. Create a profile like https://cohost.org/eloquence
2. Add a Mastodon link to your profile (it will be rendered with rel="me")
3. Add that Cohost profile to your Mastodon bio fields

### Expected behaviour

Verification succeeds, green checkmark

### Actual behaviour

Verification fails, no green checkmark

### Mastodon instance

social.coop

### Mastodon version

4.2.0

### Technical details

I've created a little Gist here (https://gist.github.com/eloquence/740520d8e377745545ebcc9ecce3e4ab) to see if I can simulate Mastodon's verification code locally. I'm not able to reproduce the issue this way - Nokogiri does find the expected backlink if I run my gist.

**Comment by trwnh** [comment]
possible duplicate of #27001?

**Comment by MadokVaur** [comment]
At "rel="me nofollow noopener" target="_blank">social.coop/@eloquence</a>" what effect does the 'noopener' keyword have?

"The noopener keyword for the [rel](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel) attribute of the [<a>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a), [<area>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/area), and [<form>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form) elements instructs the browser to navigate to the target resource without granting the new browsing context access to the document that opened it"

All I can say is that I did *not* use that keyword on my own web site and the link works fine

**Comment by eloquence** [comment]
@trwnh Maybe, though the reporter said they no longer experience the issue. I hope this one is helpful since it provides exact steps to reproduce.

**Comment by eloquence** [comment]
@MadokVaur I don't have control over Cohost's HTML, but it shouldn't matter, i.e. if that's breaking verification, it's a bug. In any case, it's not clear to me that's what's causing it - as I noted, running similar verification code locally succeeds.

**Comment by MadokVaur** [comment]
Yeah. Reason I asked was because one of my two photography web sites (Smugmug) where I have html-edit control allowed the <a ... > tag but stripped out the rel=me guts that did the actual work

Here I'm wondering if Cohost is trying to be "helpful". Can you edit that out and save?

**Comment by nightpool** [comment]
If the verification code works locally, then there must be something with either social.coop's code or host that is causing the issue, so I don't think there's any way for us to troubleshoot this further. You'll have to reach out to someone with admin permissions to social.coop for more information. Happy to reopen if this does prove to be a bug in Mastodon somehow but your gist seems to indicate pretty conclusively that it's not. My guess is that cohost just blocks social.coop's IP or user-agent.

**Comment by eloquence** [comment]
@nightpool I don't think it's specific to social.coop. What caused me to troubleshoot this in the first place is that another user on the gamedev.lgbt instance was experiencing the same issue, and I attempted to reproduce it. On both instances, other verifications are performed successfully.
